src/makeCalculator.js

```diff
@@ -4,7 +4,64 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    add(value) {
+      this.result += value;
+
+      return this;
+    },
+
+    subtract(value) {
+      this.result -= value;
+
+      return this;
+    },
+
+    multiply(value) {
+      this.result *= value;
+
+      return this;
+    },
+
+    divide(value) {
+      if (value !== 0) {
+        this.result /= value;
+      } else {
+        throw new Error("Can't divide by 0!");
+      }
+
+      return this;
+    },
+
+    operate(operation, value) {
+      switch (operation) {
+        case this.add:
+          this.add(value);
+          break;
+        case this.subtract:
+          this.subtract(value);
+          break;
+        case this.multiply:
+          this.multiply(value);
+          break;
+        case this.divide:
+          this.divide(value);
+          break;
+        default:
+          throw new Error('Unknown operation');
+      }
+
+      return this;
+    },
```

Comment:
In the `operate` method, you are checking if the operation passed is one of the calculator's own methods. However, the task requirements specify that any function can be passed to `operate` as a callback. Therefore, you should not limit the callback to only the calculator's own methods. Instead, you should check if the callback is a function and then call it. Here is a suggestion: 

js
if (typeof operation === 'function') {
  operation.call(this, value);
} else {
  throw new Error('Callback is not a function');
}
```